Lib/dataset.py

```diff
@@ -1889,10 +1892,15 @@ def write(self, var, attributes=None, axes=None, extbounds=None, id=None,
 
         # Make var an AbstractVariable
         if dtype is None and typecode is not None:
-            dtype = typeconv.convtypecode2(typecode)
+#            dtype = typeconv.convtypecode2(typecode)
+            dtype = typecode
         typecode = dtype
         if typecode is not None and var.dtype.char != typecode:
             var = var.astype(typecode)
+        if var.dtype.char == 'l' and Cdunif.CdunifGetNCFLAGS("classic"):
+            var = var.astype(numpy.int32)
```

Comment:
This is important for the attribute.  I cannot write int64 attribute sin a `netCDF classic format`.   Data and _FillValue needs to be int64 but not `integer` attributes in this case.
Yes test fails if this is not there!